src/shared/Microsoft.Git.CredentialManager/Authentication/MicrosoftAuthentication.cs

```diff
@@ -9,6 +9,10 @@
 using Microsoft.Identity.Client;
 using Microsoft.Identity.Client.Extensions.Msal;
 
+#if NETFRAMEWORK
+using Microsoft.Identity.Client.Desktop;
+#endif
```

Comment:
Is this going to be supported in .NET 5? Does adding this feature add more debt towards upgrading to .NET 5?

Reply:
For a possible future move to .NET 5 on Windows, we'd be using the `net5.0-windows*` TFMs, of which the MSAL library includes the WAM pieces already.

The `MS.ID.Client.Desktop` package is there to bring those Windows-only dependencies in for .NET Framework (or .NET Core 3.x and below) without adding them to the core, x-plat `MS.ID.Client` library. Several of their customers had hard blockers on introducing things like a dependency on WPF or WinRT APIs in say, a use of MSAL in an ASP.NET application.

I've been round the houses on getting us on .NET 5 on Windows, but it's fraught with problems. Primarily around: a) trimming/size of the resulting application (bigger than the rest of G4W!), b) multi-targeting and down-level Windows support (win7+8.x), c) UI 'helper' assemblies.

cc: @bgavrilMS

Reply:
Yeah, ideally WAM support would be inside MSAL on all frameworks, however the depedencies on nuget packages that call into WinRT makes it impossible. In MSAL 5 we're thinking of making WAM the default option, and thus we'd need to separate MSAL into an MSAL for public client and MSAL for confidential client.